src/rt-app_args.c

```diff
@@ -60,11 +63,54 @@ struct option long_args[] = {
 	{0,		0,		0,	0}
 };
 
+static char* main_install_path = NULL;
+
+char const* get_shared_library_base_path()
+{
+	return main_install_path;
+}
+
 void
 parse_command_line(int argc, char **argv, rtapp_options_t *opts)
 {
 	struct stat config_file_stat;
 	int c;
+	int full_path_len;
+	int main_install_path_len;
+
+	static const char shared_lib_dir_name[] = "/../rtapp_function/";
+
+	/* get the rt-app base directory full name */
+	char* full_path = malloc(PATH_MAX);
+	int index;
+
+	/* get the program invocation string and extract the base directory content */
+	if (!realpath(argv[0], full_path))
+	{
+		log_error("failed to get rt-app base path: %s", strerror(errno));
+		exit(0);
+	}
+
+	for(index=strlen(full_path)-1; index>0; index--)
+	{
+		if(full_path[index]=='/')
+		{
+			full_path[index]='\0';
+			break;
+		}
+	}
+
+	full_path_len = strlen(full_path);
+
+	main_install_path_len = full_path_len + sizeof(shared_lib_dir_name) + 1;
+	main_install_path = malloc(main_install_path_len);
+	if (!main_install_path)
+	{
+		log_error("failed to allocate %d bytes: %s", main_install_path_len, strerror(errno));
+		exit(0);
+	}
+
+    snprintf(main_install_path,main_install_path_len,"%s%s", full_path, shared_lib_dir_name);
```

Comment:
Guess we should check for errors?
Also please align properly with a tab.

Reply:
done